server/src/guest/guest.controller.ts

```diff
@@ -32,16 +34,21 @@ export class GuestController {
 	@Get(':uuid')
 	async read(
 		@Param('uuid') uuid: string,
-	): Promise<guestData | rsvpData | primaryData> {
+	): Promise<
+		| guestData
+		| (rsvpData & contactData & primaryData)
```

Comment:
I don't really like this as our type for the Promise. It's almost not useful?

Reply:
As in - how do I act on what I'll get back?

Reply:
Sorry to comment spam - 
Specifically, I don't like the combination of (rsvpData & contactData & primaryData) OR (rsvpData & primaryData)

The former will contain the latter, which feels a bit redundant. If we just get this down to guestData | (the combined type thing), that'll probably be fine